### ISSUE TITLE
network: automatically set accept_ra when using SLAAC

### ISSUE THREAD

**Issue by Ulexus** [siderolabs/talos#3841]
## Feature Request

If a user is receiving their IPv6 address via SLAAC, it is very likely that they are _also_ expecting to receive routes via Router Advertisements.  The default kernel behaviour, however, is to ignore router advertisements if ip_forwarding is enabled.  This is done because _usually_ when ip_forwarding is enabled, the node is acting as a router, so such routes will be otherwise configured.

In Kubernetes, however, while the node is technically a router, people expect to operate it as a node.  Thus, we should also accept router advertisements on any host interface which obtains a SLAAC address.

In order to do this, we need to set `net.ipv6.conf.<interface_name>.accept_ra` to `2` instead of the default `1`.

**Comment by anthr76** [comment]
Any chance we can pick this up for v0.14?

**Comment by smira** [comment]
@anthr76 we don't have this scheduled for 0.14 most probably, but this should be updateable via `.machine.sysctls` in the machine configuration

**Comment by flokli** [comment]
This is useful so kubernetes nodes themselves also have IPv6 connectivity between each other, especially in IPv6-only deployments. Same applies to https://github.com/talos-systems/talos/issues/3840.

Also note some CNIs (such as flannel) seem to require dualstack on the nodes to provide dualstack for pods.

**Comment by telmich** [comment]
Party!

**Comment by sergelogvinov** [comment]
Unfortunately is not working.

```shell
# sysctl net.ipv6.conf.eth1.accept_ra
net.ipv6.conf.eth1.accept_ra = 1
# sysctl net.ipv6.conf.all.accept_ra
net.ipv6.conf.all.accept_ra = 1
# sysctl net.ipv6.conf.default.accept_ra
net.ipv6.conf.default.accept_ra = 2
```

Maybe it will work directly for special interface  ```net.ipv6.conf.eth0.accept_ra=2```

**Comment by Ulexus** [comment]
I wonder why it would work for `net.ipv6.conf.default.forwarding` but not `net.ipv6.conf.default.accept_ra`...

**Comment by sergelogvinov** [comment]
Proposal:

Depend on cloud providers (platform) we need to disable/enable slaac to the interface.

for example add an option to the ```LinkSpecs.net.talos.dev``` resource:

https://github.com/siderolabs/talos/blob/91a49c4e7c0495794a991ab7742cd06339ce072d/pkg/machinery/resources/network/link_spec.go#L23-L34

like:

```go
	// IPv6 accept router advertisement.
	AcceptRA bool `yaml:"acceptRa,omitempty"`
```

AcceptRA is boolean, and false => accept_ra=0, true => accept_ra=3, by default AcceptRA will be ```true``` for all non local interfaces (!lo,!dummy)

To override the default value we can add the param in machine-config.

```yaml
  network:
    interfaces:
      - interface: eth1
        disabledRA: true
```

**Comment by smira** [comment]
https://github.com/systemd/systemd/blob/bc33789a06e5a727fa4662b0dfcbe02ef7e46687/man/systemd.network.xml#L727-L753

**Comment by smira** [comment]
RA implementation in Go: https://github.com/mdlayher/ndp

**Comment by maxpain** [comment]
Any updates?

**Comment by flokli** [comment]
Still relevant, ipv6 isn't going away.

**Comment by dedene** [comment]
Still absolutely relevant!

**Comment by zargony** [comment]
Same here (SBC nodes in homelab with SLAAC). I need to set `net.ipv6.conf.eth0.accept_ra: '2'` in machineconfig, otherwise the IPv6 default route isn't set and hosts and pods don't have IPv6 internet access. See also #5632.

**Comment by james-callahan** [comment]
Running into this with Talos on AWS using calico IPv6.

**Comment by james-callahan** [comment]
This is still an issue.

**Comment by telmich** [comment]
I very much second this, as it's the main reason for not being able to use talos.

**Comment by rouke-broersma** [comment]
Anyone know why #4939 did not fix the issue? Can we set a static gateway route in talos as a workaround or does that not fix this issue?

**Comment by telmich** [comment]
Static gateways don't really work with SLAAC enabled networks, kind of defeats the purpose. It's like using DHCP, but adding routes manually on each node.

**Comment by rouke-broersma** [comment]
@telmich I know it's not ideal, I'm asking if that works as a workaround (as in has anyone tested this).

**Comment by sergelogvinov** [comment]
> [@telmich](https://github.com/telmich) I know it's not ideal, I'm asking if that works as a workaround (as in has anyone tested this).

Yep, you can use static ipv6 route, just change the metric value.

```
    interfaces:
      - interface: eth0
        ...
        routes:
          - network: ::/0
            gateway: fe80::1234:5678:9abc
            metric: 2048
```

**Comment by Ulexus** [comment]
Yes, you _can_ set IPv6 up completely statically, routes, addresses, and all.  But again, for SLAAC networks, that is very much not the point.  Moreover, where networks are _dynamic_, such as EVPN/VXLAN networks, such routes cannot be known in advance.
